cmd/xl-zones.go

```diff
@@ -1319,6 +1284,50 @@ func (z *xlZones) HealBucket(ctx context.Context, bucket string, dryRun, remove
 	return r, nil
 }
 
+func (z *xlZones) Walk(ctx context.Context, bucket, prefix string, doneCh chan struct{}) <-chan ObjectInfo {
+	var zonesEntryChs [][]FileInfoCh
+
+	recursive := true
+	for _, zone := range z.zones {
+		zonesEntryChs = append(zonesEntryChs,
+			zone.startMergeWalks(ctx, bucket, prefix, "", recursive, doneCh))
+	}
+
+	var zoneDrivesPerSet []int
+	for _, zone := range z.zones {
+		zoneDrivesPerSet = append(zoneDrivesPerSet, zone.drivesPerSet)
+	}
+
+	var zonesEntriesInfos [][]FileInfo
+	var zonesEntriesValid [][]bool
+	for _, entryChs := range zonesEntryChs {
+		zonesEntriesInfos = append(zonesEntriesInfos, make([]FileInfo, len(entryChs)))
+		zonesEntriesValid = append(zonesEntriesValid, make([]bool, len(entryChs)))
+	}
+
+	objInfoCh := make(chan ObjectInfo)
+
+	go func() {
+		defer close(objInfoCh)
+
+		for {
+			entry, quorumCount, zoneIndex, ok := leastEntryZone(zonesEntryChs,
```

Comment:
Same - should it be possible to cancel the walk?!

Reply:
doneCh is already passed

Reply:
Already fixed